ThunderstoreCLI/Utils/Spinner.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using static Crayon.Output;
+
+namespace ThunderstoreCLI
+{
+    public class ProgressSpinner
+    {
+        private int _lastSeenCompleted = 0;
+        private ushort _spinIndex = 0;
+        private string[] _spinChars = { "|", "/", "-", "\\" };
+        private string _label;
+        private readonly Task[] _tasks;
+
+        public ProgressSpinner(string label, Task[] tasks)
+        {
+            if (tasks.Length == 0)
+            {
+                throw new SpinnerException("Task list can't be empty");
+            }
+
+            _label = label;
+            _tasks = tasks;
+        }
+
+        public async Task Start()
+        {
+            while (true)
+            {
+                if (_tasks.Any(x => x.IsFaulted))
+                {
+                    Write.Empty();
+                    throw new SpinnerException("Some of the tasks have faulted");
```

Comment:
Is there a way to include more exception details in this error about what _actually_ failed? I know in python you can do `throw Exception from some_other_exception`, but not sure if there's something similar here. Either way as I understand it, this would currently hide any errors that happen in tasks without producing any output about said errors, is this correct?

Reply:
In the [current use case](https://github.com/thunderstore-io/thunderstore-cli/blob/fix-console-cursor-usage/ThunderstoreCLI/Commands/PublishCommand.cs#L232-L311) the tasks themselves output the error messages, and the `SpinnerException` thrown here is used only as a signal to abort the upload, and its message is ignored.

I guess this can be changed so that instead of using `_tasks.any`, we iterate over the tasks and use the exception message of the first faulted task as the exception message in SpinnerException. But again, in the only use case we currently have, the message is ignored to avoid showing the same exceptions twice.

Reply:
Gotcha, so let's create an issue about this and merge it as-is.